lib/locale.ts

```diff
@@ -3,11 +3,41 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 
+declare global {
+	// eslint-disable-next-line camelcase, no-var
+	var _nc_l10n_locale: string | undefined
+	// eslint-disable-next-line camelcase, no-var
+	var _nc_l10n_language: string | undefined
+}
+
 /**
  * Returns the user's locale
+ * @example 'de_DE'
  */
 export function getLocale(): string {
-	return document.documentElement.dataset.locale || 'en'
+	// Web-browser runtime
+	if (typeof document !== 'undefined' && document.documentElement.dataset.locale) {
+		return document.documentElement.dataset.locale
+	}
+
+	// Node.js runtime
+	if ('_nc_l10n_locale' in globalThis && globalThis._nc_l10n_locale) {
+		return globalThis._nc_l10n_locale
+	}
+
+	// Fallback to English
+	return 'en'
```

Comment:
`globalThis` always works in all contexts.


```suggestion
	return globalThis._nc_l10n_locale ?? 'en'
```